### ISSUE TITLE
Code with errors makes TeXstudio shut down

### ISSUE THREAD

**Issue by LorentsL** [texstudio-org/texstudio#3441]
### Environment
- TeXstudio: 4.7.1
- Qt: 6.6.1
- OS: Windows 10/11
- TeX distribution: texlive

### Expected behavior
A compilation issue reported in the "Log" section. The program keeps running.

### Actual behavior
The syntax error(s) in the code causes TeXstudio to shut down. On Windows 10 it closes immediately without warning. Open documents are saved in their current state. (Unsaved drafts are lost.) On Windows 11 the following pop-up is shown.
![texstudio_crash_3](https://github.com/texstudio-org/texstudio/assets/120018586/741364ca-6aa9-4de5-98d9-ae918b61f561)

### How to reproduce

**Edit:** Copy the code below, save it in a tex-file, and compile in TeXstudio with pdflatex.

I have shortened the code as much as I can, so that the crash still happens. However, removing more things seems, in general, to stop the problem from occuring. The immediate cause of the crash is in the first frame, where I forgot to close the "columns"-environment. Without these two lines, the code compiles without error. However, there is at least one other syntax problem near the end of the document, where a closing bracket is misplaced, as noted in code comments.

```
\documentclass{beamer}
\usepackage{tikz}
\usepackage[absolute,overlay]{textpos}
\begin{document}

\begin{frame}{}
	
	\begin{columns}	
		\column{0.5\textwidth} % Without these two lines, the code compiles. With these lines, the program shuts down
		
\end{frame}

\begin{frame}{}
    \begin{block}{Theorem}
    	A graph with a cutvertex can not be optimal.
    \end{block}
    \onslide<1->{{Proof idea:}\vskip 2mm} 
    \only<1>{
	    \begin{tikzpicture}[x=0.7cm, y=0.7cm]
	        \node at (3,0) {$G$};
	        \filldraw[thick] (0,0) node (1){} -- (2,1) circle (3pt) node (a){}-- (4,1) circle (3pt) node (b){} -- (6,0) node (3){};
	        \filldraw[thick] (0,2) node (2){} -- (a.center) (6,2) node (4){}-- (b.center);
	    \end{tikzpicture}
    }
    \only<2-4>{
	    \begin{tikzpicture}[x=0.7cm, y=0.7cm]
	        \node at (3,0) {$G'$};
	        \filldraw[thick] (0,0) node (1){} -- (2,0.8) circle (3pt) node (a){}-- (4,1) circle (3pt) node (b){} -- (6,0) node (3){};
	        \filldraw[thick] (0,2) node (2){} -- (b.center) (6,2) node (4){}-- (b.center);
	    \end{tikzpicture}
    }\vskip 2mm
    
    \begin{textblock*}{5.2cm}(6.6cm,2.8cm)
	    \onslide<3->{
		    \begin{alertblock}{\small{Layout graph when $m=n+1$}}
		        %\vskip -3mm 
		        \hspace{7mm}
		        \begin{tikzpicture}
		           \filldraw[thick] (0,0) circle (1.4pt) node (a){} --(0,1) circle (1.4pt) node (b){}; 
		           \draw[thick] (a.center) edge[out=180,in=180] (b.center);
		           \draw[thick] (a.center) edge[out=0,in=0] (b.center);
		        \end{tikzpicture}
		        %\vspace{-3 mm}       
		    \end{alertblock}
	    }
    \end{textblock*} 
\end{frame}

\begin{frame}{}
	\vskip -4mm
	\begin{columns}
		\column{0.76\textwidth}
		\small{
			\begin{block}{Theorem}
				Given $n$ vertices and $n+2$ edges, there is a unique optimal graph, independently of $p$.
				\onslide<2->{ The graph has
					\begin{itemize}\itemsep=-0.4mm
						\item six chains
				}
						\onslide<3->{\item of as equal length as possible}
						\onslide<4->{\item arranged as the complete graph $K^4$.}
					\end{itemize}
			\onslide<5->{Furthermore:
				\begin{itemize}\itemsep=-0.4mm
					\onslide<6->{\item If two chains are longer,\\ they should be opposite each other.}\vskip 2pt
					\onslide<7->{\item If three chains are longer,\\ they should form a simple path.}\vskip 2pt
					\onslide<8->{\item If two chains are shorter,\\they should be opposite each other.} \vspace{2pt}
				\end{itemize}
			}
			\end{block}
		}
		\column{0.28\textwidth}
		\begin{textblock*}{10cm}(9.45cm,3cm)
		\only<4>{
			\begin{tikzpicture}[x=1.1cm, y=1.1cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a.center)--(c.center) (b.center)--(d.center);
			\end{tikzpicture}
		}
		\only<5>{
			\begin{tikzpicture}[x=1.2cm, y=1.2cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a.center)--(c.center) (a) circle (3pt) (b.center)--(d.center);
			\end{tikzpicture}
		}
		\only<6>{
			\begin{tikzpicture}[x=1.2cm, y=1.2cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a.center)--(c.center) (a) circle (3pt) (a) circle (3pt) (b.center)--(d.center);
			\end{tikzpicture}
		}
		\only<7>{
			\begin{tikzpicture}[x=1.2cm, y=1.2cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a.center)--(c.center) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (b.center)--(d.center);
			\end{tikzpicture}
	}
		\only<8>{
			\begin{tikzpicture}[x=1.2cm, y=1.2cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a.center)--(c.center) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (b.center)--(d.center);
			\end{tikzpicture}
	}
		\only<9>{
			\begin{tikzpicture}[x=1.2cm, y=1.2cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (a.center)-- (0.8,1.2) circle (3pt) --(c.center) (b.center)--(d.center);
			\end{tikzpicture}
		}
		\only<10>{
			\begin{tikzpicture}[x=1.2cm, y=1.2cm]
				\filldraw[thick] (0,0) circle (3pt) node (a){} -- (2.1,0) circle (3pt) node (b){} -- (2.1,3) circle (3pt) node (c){} -- (0,3) circle (3pt) node (d){} (a.center)--(d.center) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (a) circle (3pt) (a.center)-- (0.8,1.2) circle (3pt) --(c.center) (b.center)-- (0.8,1.9) circle (3pt)--(d.center);
			\end{tikzpicture}
		}
		\end{textblock*}
	\end{columns}
\end{frame}

\begin{frame}{Further work:}
	\begin{block}{}
		For ${(n,n+3)}$\textbf{-graphs} there are two simple, 3-regular layout graphs.
	\end{block}
	\vskip 4mm
	\onslide<2->{
		\begin{tikzpicture}[x=0.9cm,y=0.9cm]
			\filldraw (1,0) circle (2pt) node (a){}--(0.5,0.87) circle (2pt) node (b){}--(-0.5,0.87) circle (2pt) node (c){} -- (-1,0) circle (2pt) node (d){} -- (-0.5,-0.87) circle (2pt) node (e){} -- (0.5,-0.87) circle (2pt) node (f){} -- (a.center) (a.center) --(d.center) (b.center)--(e.center) (c.center)--(f.center);
			\node at (-1.2,0.9) {$K_{3,3}$};
	\end{tikzpicture}\hspace{2mm}
	\begin{tikzpicture}[x=0.9cm,y=0.9cm]
		\filldraw (1,0) circle (2pt) node (a){}--(0.5,0.87) circle (2pt) node (b){}--(-0.5,0.87) circle (2pt) node (c){} -- (-1,0) circle (2pt) node (d){} -- (-0.5,-0.87) circle (2pt) node (e){} -- (0.5,-0.87) circle (2pt) node (f){} -- (a.center) (a.center) --(c.center) (b.center)--(e.center) (d.center)--(f.center);
		\node at (-1.2,0.9) {$H_6$};
	\end{tikzpicture}
	}
	\begin{textblock*}{4.4cm}(7.3cm,2.3cm)
		
	\onslide<3->{
		\begin{block}{Conjecture for $m = n + 3$}
			The optimal graph is unique, $p$-independent, based upon $K_{3,3}$ and has E.A.P. chains.
	}				% Syntax error here, lines swapped
		\end{block} % Syntax error here, lines swapped
	
	\end{textblock*}
	\onslide<4->{
	\vskip 4mm
		\begin{block}{}
			For ${(n,n+4)}$\textbf{-graphs} there are four simple, 3-regular layout graphs, e.g.\ the Wagner graph and the graph of the cube.
		\end{block}
	}
	\onslide<5>{
		\begin{block}{Conjecture for $m = n + 4$}
			The $(n,m,p)$-optimal graphs are not the same for large $p$ compared to for small $p$.
	  	\end{block}
	}
\end{frame}
\end{document}
```

**Comment by sunderme** [comment]
I can not reproduce a crash on win11.
Do you have any non-defaupt settings ?
Activated completion files,.etc ?

**Comment by octaeder** [comment]
I couldn't reproduce with current dev and Win10. When I copied the code into a new editor (Ctrl+N) the parser/highlighter immediately showed the first issue:

![grafik](https://github.com/texstudio-org/texstudio/assets/102688820/8eb3f204-03ee-4c13-a8c3-36a986e84456)

How do you compile? I used pdflatex and after a while a list of errors appeared:

![grafik](https://github.com/texstudio-org/texstudio/assets/102688820/03f6ac11-2a6f-478a-bf7c-5bbc2c2a8181)

**Comment by LorentsL** [comment]
It seems to me that the .tex file has to be saved on disk first. Then, when compiling, TeXstudio crashes on both systems that I have available.

I'm using pdflatex, and have not meddled with the settings further than visual appearance (dark theme, etc.).

Is there any log-file that could be helpful?

**Comment by octaeder** [comment]
Thanks for the update. I use name C:/msys64/home/WDAG/test_issue#3441.tex but there is no crash in my case.
Can you try with default settings? Rename folder `%appdata%/texstudio` after closing txs.

**Comment by octaeder** [comment]
now I tried with installed versions 4.6.3 and 4.7.1, but still not reproducible.
waiting for your result with defaults.

**Comment by LorentsL** [comment]
Hi octaeder! I restored the settings to default by renaming the folder (thanks for the instructions). I even created folders to save the tex-file (which I copied from the report above) in the exact same path as you did. Pressed compile, and after ~5 seconds the program shut down!

**Comment by sunderme** [comment]
screencast?

**Comment by LorentsL** [comment]
screencast:
https://github.com/texstudio-org/texstudio/assets/120018586/e8ba0bba-4bb5-41fc-a97b-84cb18ae1585

**Comment by octaeder** [comment]
Sorry, I still can't reproduce with current dev.

**Comment by sunderme** [comment]
thismlooks like loading of the log file causes the issue.
Can you provide all generated files from the first document.tex ? document.log etc.

**Comment by muzimuzhi** [comment]
> screencast: https://github.com/texstudio-org/texstudio/assets/120018586/e8ba0bba-4bb5-41fc-a97b-84cb18ae1585

@LorentsL BTW, you can click the button on upper right corner of a fenced code block to copy its content. See https://stackoverflow.com/a/67478530

**Comment by LorentsL** [comment]
Now, _I_ had a hard time reproducing the crash! But after a couple of tries, the program crashed again. Here are the generated files. (Most file types were not supported, so I zipped everything.)
[doc.zip](https://github.com/texstudio-org/texstudio/files/13765607/doc.zip)

**Comment by sunderme** [comment]
fixed with c0fef7c2ef5e4da4e61b3b866602797bf8e1f52e

**Comment by LorentsL** [comment]
Nice. And merry Christmas!
